raft/README.md

```diff
@@ -5,6 +5,14 @@ The state machine is kept in sync through the use of a replicated log.
 For more details on Raft, see "In Search of an Understandable Consensus Algorithm"
 (https://ramcloud.stanford.edu/raft.pdf) by Diego Ongaro and John Ousterhout.
 
+This Raft library is stable and feature completed. It is **the most widely used** (as of 2016) Raft library that serves tens of thousands clusters each day. It powers distributed systems like etcd, Kubernetes, Docker Swarm, Cloud Foundry Diego, CockroachDB, TiDB, Project Calico, Flannel and more.
+
+Most Raft implementations has a monolithic design, which includes storage, messaging serialization, transport. This Raft library takes a very different approach by only implementing the core algorithm part of raft. The design philosophy of this raft library is minimalism, flexibility, determinism, and performance.
+
+To minimize the library as well as provide flexibility, it only implements the Raft algorithm part and leave the network IO and disk IO parts to its users. Users need to implement their own transportation layer to enable actual message passing between Raft peers over wire. Also users need to implement their own storage layer to persist Raft logs and states.
```

Comment:
To keep the codebase small as well as provide flexibility, the library only implements the Raft algorithm; both network and disk IO are left to the user. Library users must implement their own transportation layer for message passing between Raft peers over the wire. Similarly, users must implement their own storage layer to persist the Raft log and state.